src/puppet.js

```diff
@@ -622,40 +642,18 @@
 
   Puppet.prototype.handleRemoteChange = function (data, url, method) {
     var patches = JSON.parse(data || '[]'); // fault tolerance - empty response string should be treated as empty patch array
-    var that = this;
 
     if (this.onPatchReceived) {
         this.onPatchReceived(data, url, method);
     }
 
-    if (!this.observer) {
-      return; //ignore remote change if we are not watching anymore
-    }
+    // apply only if we're still watching
+    if (this.observer) {
```

Comment:
Personally I prefer "return fast" in simple cases, as then I do not have to read entire method, and search for applicable `else`